### ISSUE TITLE
Support for json format

### ISSUE THREAD

**Issue by cpmarvin** [mre/kafka-influxdb#36]
Hello,

Would be nice to have an encoder to support json and be able to choose what keys are tags or values and the time field.

so for example in below json  : 

```json
{
 "interface_id": 6939,
 "bytes": 13780000,
 "interface_name": "eth0",
 "time": " 1470920344.173186
}
```

we can choose interface_id, interface_name as tags and bytes as values resulting
example_measuarment,interface_id=6939,interface_name="eth0" bytes= 13780000 1470920344.173186

**Comment by mre** [comment]
Hi @cpmarvin,
please take a look at https://github.com/mre/kafka-influxdb/tree/master/kafka_influxdb/encoder.
There's a little description on how to write your own encoder.
You could start with the json encoder and modify it to your needs.

I'm still thinking about a smarter approach to do this. We could add the tag names into the config, but I don't know if this covers all use-cases...

**Comment by cpmarvin** [comment]
hi @mre 

Thx for that. 
Yes the ideea was to have a generic one and have config flags for tags, value and time fields to match against the json.

**Comment by mre** [comment]
So the only thing to do would be adding an optional `tags` field in the config and pass it to the encoder?
Maybe we could also make it a bit more generic and have an `encoder_params` section in the config that gets passed to the encoder?

**Comment by cpmarvin** [comment]
i would say , 3 files . one with tags ( multiple values ) one for value and of for timestamp ( if not defined use current time ) , using  encoder_params.

**Comment by mre** [comment]
Yeah, sounds reasonable. I like your idea.
Not sure when I will come around to implement it, though. So feel free to grab this task if you like. I can support you if you like. :smiley:

**Comment by mre** [comment]
Giving this some more thought, I'm not sure what we will do with the collectd graphite encoder.
It does not have a key-value structure like collectd-json.

Metrics look like this:
```
myhost.load.load.shortterm 0.05 1436357630
```

I would love to have one `encoder_params` section in the config that is compatible with all encoders, but I can't find a good way to make it work. Open for suggestions.

**Comment by mre** [comment]
@kbr rewrote the graphite encoder with tag support. See feature/encoder_rewrite_v2.
I'll do the same for the json encoder now. Will push my changes to the same branch and ping you again once this is ready. Stay tuned.

**Comment by mre** [comment]
> I'll do the same for the json encoder now. 

Err.. no, I didn't. Sorry for the radio silence. Due to other projects I won't work on kafka-influxdb anymore. That's why I'm closing this for now. In any case, if someone wants to tackle this, let me know and we can reopen.
